enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,288 @@
+---
+title: aws-load-balancer-type
+authors:
+  - "@miheer"
+reviewers:
+  - "@wking"  
+  - "@Miciah"
+  - "@frobware"
+  - "@sdodson"
+approvers:
+  - "@wking"  
+  - "@sdodson"
+  - "@frobware"
+api-approvers:
+  - "@deads"
+  - "@Miciah"
+tracking-link:
+  - "https://issues.redhat.com/browse/NE-942"
+creation-date: 2022-06-09
+last-updated: 2019-06-09   
+status: yet to be implemented
+superseded-by:
+  - ""
+---
+
+# Allow Users to specify Load Balancer Type during installation for AWS
+
+## Release Signoff Checklist
+
+- [ x ] Enhancement is `implementable`
+- [ x ] Design details are appropriately documented from clear requirements
+- [ x ] Test plan is defined
+- [ x ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The users would be able to specify the load balancer type either to elb or nlb during installation. 
+This will be used to deploy default ingress controller with user-specified load balancer service for OpenShift clusters
+running on AWS.
+Currently by default CLB/ELB is set by default by the cluster-ingress-operator.
+
+
+## Motivation
+
+Currently the default load balancer type is set to CLB/ELB. One can set default load balancer during installation using [this](https://docs.openshift.com/container-platform/4.10/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws-network-load-balancer.html#nw-aws-nlb-new-cluster_configuring-ingress-cluster-traffic-aws-network-load-balancer).
+However when the default IngressController CR is deleted, default ingress controller is recreated but as an CLB/ELB.
+This  behavior is expected as of now as the ingress operator by default creates CLB for default ingress controller. 
+Also if the CR is deleted then the operator wont be able check the information whether NLB is needed or not. 
+So, we need to store this information in the cluster object of ingress.
+
+### Goals
+
+As an administrator, I would like to specify the load balancer type in AWS either to NLB or CLB/ELB during installation.
+As an user/administrator, when an ingress controller CR is deleted then I would like the ingress controller to be recreated with the load balancer mentioned during installation.
+In short I would like the persistence of the load balancer type in the default ingress controller.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We will be adding a field awsLBType in the install-config for the users to provide a load balancer either to NLB or CLB/ELB
+during installation.
+
+### Workflow Description
+
+The installer will accept a LB type from user and then we would set type of LB in the following object.
```

Comment:
So, would AWSLoadBalancerType in the Ingress CR  https://github.com/openshift/api/blob/master/operator/v1/types_ingress.go#L462 be updated with the value provided in the new field proposed for the install-config?